### ISSUE TITLE
Overflow hidden

### ISSUE THREAD

**Issue by lohek** [nkbt/react-collapse#163]
Hi !
I have issue when the content of Collapse change, a `overflow: hidden` is added with dynamic height.
While it would only take the `height: auto`.

![](http://g.recordit.co/ZignxDEfBg.gif)

I tried with `Collapse` and `UnmountClosed` but i have the same behavior.

```
<div>
  {items.map((item, i) => [
    <div
      key={`${i}-header`} // eslint-disable-line react/no-array-index-key
      onClick={() => this.setState({ openedIndex: i })}
    >
      {item.header}
    </div>,
    <Collapse
      key={`${i}-body`} // eslint-disable-line react/no-array-index-key
      isOpened={i === openedIndex}
      springConfig={{ stiffness: 250 }}
    >
      {React.cloneElement(item.body, this.props)}
    </Collapse>,
  ])}
</div>
```

Thanks !

**Comment by nkbt** [comment]
Hidden must be applied to the container element, otherwise collapse is useless. After animation is compketed hight is back to auto.

I cannot see any techically possible way to avoid it.

**Comment by bradennapier** [comment]
Hmm - when doing a nested collapse it overflow hidden and the content therefore never shows up at all.

**Comment by jonom** [comment]
I think I'm seeing the same issue. It seems if a nested collapse element is set to be expanded on mount, it will have a specific height set, and overflow set to hidden. This means elements such as drop-down menus within the Collapse can get clipped. To correct the issue I need to collapse the element and re-expand it.

**On mount:**
```html
<div class="ReactCollapse--collapse" style="height: 27px; overflow: hidden;">
```
**After collapsing and re-expanding:** (these are the values the element should have on mount)
```html
<div class="ReactCollapse--collapse" style="height: auto;">
```

**Comment by jonom** [comment]
I was experimenting with a fix, I don't know if this is a good idea but changing a little bit in `componentWillReciveProps()` _seemed_ to fix things for me.

**Before**
```js
} else if (this.state.currentState === IDLING && (nextProps.isOpened || this.props.isOpened)) {
    this.setState({ currentState: WAITING });
}
```
**After**
```js
} else if (this.state.currentState === IDLING && (nextProps.isOpened !== this.props.isOpened)) {
    this.setState({ currentState: WAITING });
}
```

**Comment by nkbt** [comment]
This solution breaks animation. See comment https://github.com/nkbt/react-collapse/pull/166#issuecomment-299395765

**Comment by jonom** [comment]
Oh I should have checked for open pull requests. Look like there might be a good solution already provided.

**Comment by nkbt** [comment]
Should be fixed in `react-collapse@4.0.3`, feel free to reopen if there are more issues
